### PR TITLE
[Rule Tuning] Suspicious CertUtil Commands

### LINKED ISSUE #1179: [Rule Tuning] Encoding or Decoding Files via CertUtil
## Link to rule

https://github.com/elastic/detection-rules/blob/main/rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml

## Description

Adding

- `?urlcache`
- `?verifyctl` 
- `?decodehex`

```
process where event.type == "start" and
  (process.name : "certutil.exe" or process.pe.original_file_name == "CertUtil.exe") and 
  process.args : ("?decode", "?encode", "?urlcache", "?verifyctl", "?decodehex")
```

## References

- https://twitter.com/Moriarty_Meng/status/984380793383370752
- https://twitter.com/egre55/status/1087685529016193025
- https://docs.microsoft.com/en-us/archive/blogs/pki/basic-crl-checking-with-certutil

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

Resolves #1179

## Summary

Adding

- `?urlcache`
- `?verifyctl` 
- `?decodehex`

```
process where event.type == "start" and
  (process.name : "certutil.exe" or process.pe.original_file_name == "CertUtil.exe") and 
  process.args : ("?decode", "?encode", "?urlcache", "?verifyctl", "?decodehex")
```

## References

- https://twitter.com/Moriarty_Meng/status/984380793383370752
- https://twitter.com/egre55/status/1087685529016193025
- https://docs.microsoft.com/en-us/archive/blogs/pki/basic-crl-checking-with-certutil

### CODE REVIEW

File: rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml [hunk 1]
```diff
@@ -15,6 +15,11 @@ index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
 language = "eql"
 license = "Elastic License v2"
 name = "Encoding or Decoding Files via CertUtil"
+references = [
+    "https://twitter.com/Moriarty_Meng/status/984380793383370752",
+    "https://twitter.com/egre55/status/1087685529016193025",
+    "https://docs.microsoft.com/en-us/archive/blogs/pki/basic-crl-checking-with-certutil",
+]
```
Comment (bm11100) [inline rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml:22]:
```suggestion
references = "https://www.sysadmins.lv/blog-en/certutil-tips-and-tricks-working-with-x509-file-format.aspx"
```

File: rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml [hunk 2]
```diff
@@ -25,7 +30,7 @@ type = "eql"
 query = '''
 process where event.type == "start" and
   (process.name : "certutil.exe" or process.pe.original_file_name == "CertUtil.exe") and 
-  process.args : ("?decode", "?encode")
+  process.args : ("?decode", "?encode", "?urlcache", "?verifyctl", "?decodehex")
```
Comment (bm11100) [inline rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml:33]:
```suggestion
  process.args : ("?decode", "?encode", "?encodehex", "?decodehex")
```

Comment (bm11100) [reply rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml:33]: I like the `urlcache` and `verifyctl` commands, but they don't deal with encoding/decoding. I think they could be split into a separate rule, or we modify this one into something like `Suspicious CertUtil Commands`.  If we choose the latter, I will adjust my review.

Comment (austinsonger) [reply rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml]: @bm11100 yeah I think changing it to `Suspicious CertUtil Commands` would be better then adding another rule. 

I'm guessing to make these additional changes?

```
name = "Suspicious CertUtil Commands"
```

File: `defense_evasion_suspicious_certutil_commands.toml`

Comment (austinsonger) [reply rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml]: @bm11100 how would you like me to proceed.

Comment (bm11100) [reply rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml]: Yeah, we would just need to change the rule name, file name, and description. With the ability to download files we could also add the Ingress Tool Transfer technique https://attack.mitre.org/techniques/T1105/.

Comment (austinsonger) [reply rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml]: Okay I changed the rule name, file name, and added the new technique. What would you like the description be?

Comment (bm11100) [reply rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml]: no need to eliminate the original technique as it is still valid. 
```
[[rule.threat]]
framework = "MITRE ATT&CK"
[[rule.threat.technique]]
id = "T1140"
reference = "https://attack.mitre.org/techniques/T1140/"
name = "Deobfuscate/Decode Files or Information"


[rule.threat.tactic]
id = "TA0005"
reference = "https://attack.mitre.org/tactics/TA0005/"
name = "Defense Evasion"
```

Comment (bm11100) [reply rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml]: Description can be something like - 
```
Identifies suspicious commands being used with certutil.exe. CertUtil is a native Windows component which is part of Certificate Services. CertUtil is often abused by attackers to live off the land for stealthier command and control or data exfiltration.
```

Comment (brokensound77) [reply rules/windows/defense_evasion_encoding_or_decoding_files_via_certutil.toml]: bump: looks like this description still needs an update

**Review by brokensound77 — APPROVED** [review]

File: rules/windows/defense_evasion_suspicious_certutil_commands.toml [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
+updated_date = "2021/05/06"
 
 [rule]
 author = ["Elastic"]
 description = """
-Identifies the use of certutil.exe to encode or decode data. CertUtil is a native Windows component which is part of
-Certificate Services. CertUtil is often abused by attackers to encode or decode base64 data for stealthier command and
-control or exfiltration.
+Identifies suspicious commands being used with certutil.exe. CertUtil is a native Windows component which is part of Certificate Services. 
+CertUtil is often abused by attackers to live off the land for stealthier command and control or data exfiltration.
```
Comment (brokensound77) [inline rules/windows/defense_evasion_suspicious_certutil_commands.toml:10]:
```suggestion
Identifies suspicious commands being used with certutil.exe. CertUtil is a native Windows component which is part of 
Certificate Services. CertUtil is often abused by attackers to live off the land for stealthier command and control or 
data exfiltration.
```

**Pull request merged**